File: Project/tolometCore/src/main/java/com/akrog/tolomet/providers/WindProvider.java
package com.akrog.tolomet.providers;

import com.akrog.tolomet.Station;

import java.util.List;

public interface WindProvider {
	void refresh( Station station );
	boolean travel(Station station, long date );
	void cancel();
	int getRefresh( String code );
	String getInfoUrl( Station sta );
	String getUserUrl( Station sta );
	List<Station> downloadStations();
}

File: Project/tolometShared/src/commonMain/kotlin/com/akrog/tolomet/providers/WindProvider.kt
package com.akrog.tolomet.providers

import com.akrog.tolomet.Station

interface WindProvider {
    suspend fun refresh(station: Station)
    suspend fun travel(station: Station, date: Long): Boolean
    fun cancel()
    fun getRefresh(code: String): Int
    fun getInfoUrl(code: String): String
    fun getUserUrl(code: String): String
    suspend fun downloadStations(): List<Station>
}
